lib/Controller/FolderController.php

```diff
@@ -89,24 +91,17 @@ public function open(?int $folderId, bool $open)
     /**
      * @NoAdminRequired
      *
-     * @param string $folderName
+     * @param string   $folderName
+     * @param int|null $parent
      *
      * @return array|JSONResponse
      */
-    public function create(string $folderName)
+    public function create(string $folderName, ?int $parent = null)
     {
-        try {
-            // we need to purge deleted folders if a folder is created to
-            // prevent already exists exceptions
-            $this->folderService->purgeDeleted($this->userId, false);
-            $folder = $this->folderService->create($folderName, $this->userId);
+        $this->folderService->purgeDeleted();
```

Comment:
Sure that we want to purge for all users? This would remove even recently deleted ones which could still be restored.

Reply:
The old code doesn't honor the arguments either :sweat_smile:

Reply:
Are you sure about that?
The called `folderMapper->getToDelete($deleteOlderThan, $userId);` inside the old `purgeDeleted` seems to honor it though
https://github.com/nextcloud/news/blob/05598a7f36a7d4bfc7afb473cc7969ca5d14b55c/lib/Db/FolderMapper.php#L110-L113

Otherwise it is somehow possible to create a race condition in a multi user environment.
User1 marks a folder as deleted; User2 creates a new folder, which purges all deleted folders; U1 tries to restore the deleted folder, since the UI still allows it, which probably leads to an error, since the folderId no longer exists.